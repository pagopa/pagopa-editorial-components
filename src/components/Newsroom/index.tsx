import EContainer from '../EContainer';
import { Typography, Grid, Link, Stack } from '@mui/material';
import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt';

interface INewsroomItem {
  img: {
    src: string;
    alt: string;
  };
  title: string;
  date: {
    date: Date;
    locale?: string;
    options?: Intl.DateTimeFormatOptions;
  };
  href: {
    label: string;
    title?: string;
    link: string;
  };
}

export interface INewsroom {
  items: INewsroomItem[];
}

const Item = (props: INewsroomItem) => {
  const {
    img: { src, alt },
    date: {
      date,
      locale = 'it-IT',
      options = {
        day: '2-digit',
        month: 'long',
        year: 'numeric',
      },
    },
    title,
    href,
  } = props;
  return (
    <Grid item md={4} mb={8}>
      <img src={src} alt={alt} width="100%" />
      <Typography color="text.secondary" fontSize={16} fontWeight={400} my={2}>
        {new Intl.DateTimeFormat(locale, options).format(date)}
      </Typography>
      <Typography variant="h6">{title}</Typography>
      <Stack mt={2} direction="row" alignItems="center" color="primary.main">
        <Link
          color="primary.main"
          underline="none"
          textTransform="capitalize"
          href={href.link}
          title={href.title}
          fontSize={14}
          fontWeight={400}
        >
          {href.label}
        </Link>
        <ArrowRightAltIcon sx={{ color: 'inerith', fontSize: 18 }} />
      </Stack>
    </Grid>
  );
};

const Newsroom = (props: INewsroom) => {
  const { items } = props;
  return (
    <EContainer background="background.paper" py={16}>
      <Grid item md={3} />
      <Grid item md={9}>
        <Grid container spacing={3}>
          {items.map((item, i) => (
            <Item key={i} {...item} />
          ))}
        </Grid>
      </Grid>
    </EContainer>
  );
};

export default Newsroom;
