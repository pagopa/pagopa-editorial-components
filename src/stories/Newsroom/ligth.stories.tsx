import { type ComponentMeta, type ComponentStory } from '@storybook/react';
import Newsroom from '../../components/Newsroom';
import newsroomcover from '../assets/newsroom-cover.png';

export default {
  component: Newsroom,
  title: 'Newsroom/light',
} as ComponentMeta<typeof Newsroom>;

const Template: ComponentStory<typeof Newsroom> = (args) => (
  <Newsroom {...args} />
);

export const Standard = Template.bind({});
Standard.args = {
  items: [
    {
      img: {
        src: newsroomcover,
        alt: 'A picture showing some books',
      },
      title:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
      date: {
        date: new Date(),
      },
      href: {
        label: 'leggi',
        title: 'vai al sito',
        link: 'https://google.it',
      },
    },
    {
      img: {
        src: newsroomcover,
        alt: 'A picture showing some books',
      },
      title:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
      date: {
        date: new Date(),
      },
      href: {
        label: 'leggi',
        title: 'vai al sito',
        link: 'https://google.it',
      },
    },
    {
      img: {
        src: newsroomcover,
        alt: 'A picture showing some books',
      },
      title:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
      date: {
        date: new Date(),
      },
      href: {
        label: 'leggi',
        title: 'vai al sito',
        link: 'https://google.it',
      },
    },
    {
      img: {
        src: newsroomcover,
        alt: 'A picture showing some books',
      },
      title:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
      date: {
        date: new Date(),
      },
      href: {
        label: 'leggi',
        title: 'vai al sito',
        link: 'https://google.it',
      },
    },
    {
      img: (
        <div>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
          eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing
          elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur
          adipiscing elit, sed do eiusmod tempor.
        </div>
      ),
      title:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
      date: {
        date: new Date(),
      },
      href: {
        label: 'leggi',
        title: 'vai al sito',
        link: 'https://google.it',
      },
    },
  ],
};
