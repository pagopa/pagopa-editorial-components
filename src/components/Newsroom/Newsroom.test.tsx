import Newsroom from './index';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';

it('renders correctly standard ', () => {
  const tree = renderer
    .create(
      <Newsroom
        items={[
          {
            date: {
              date: new Date('2023-11-23T11:35:50.763Z'),
            },
            href: {
              label: 'leggi',
              link: 'https://google.it',
              title: 'vai al sito',
            },
            img: {
              alt: 'A picture showing some books',
              src: 'static/media/newsroom-cover.59549252.png',
            },
            title:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
          },
          {
            date: {
              date: new Date('2023-11-23T11:35:50.763Z'),
            },
            href: {
              label: 'leggi',
              link: 'https://google.it',
              title: 'vai al sito',
            },
            img: {
              alt: 'A picture showing some books',
              src: 'static/media/newsroom-cover.59549252.png',
            },
            title:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
          },
          {
            date: {
              date: new Date('2023-11-23T11:35:50.763Z'),
            },
            href: {
              label: 'leggi',
              link: 'https://google.it',
              title: 'vai al sito',
            },
            img: {
              alt: 'A picture showing some books',
              src: 'static/media/newsroom-cover.59549252.png',
            },
            title:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
          },
          {
            date: {
              date: new Date('2023-11-23T11:35:50.763Z'),
            },
            href: {
              label: 'leggi',
              link: 'https://google.it',
              title: 'vai al sito',
            },
            img: {
              alt: 'A picture showing some books',
              src: 'static/media/newsroom-cover.59549252.png',
            },
            title:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
          },
          {
            date: {
              date: new Date('2023-11-23T11:35:50.763Z'),
            },
            href: {
              label: 'leggi',
              link: 'https://google.it',
              title: 'vai al sito',
            },
            img: (
              <div>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor. Lorem ipsum dolor sit amet, consectetur
                adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit
                amet, consectetur adipiscing elit, sed do eiusmod tempor.
              </div>
            ),
            title:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
          },
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('All links are rendered correctly', async () => {
  const { getAllByText } = render(
    <Newsroom
      items={[
        {
          date: {
            date: new Date('2023-11-23T11:35:50.763Z'),
          },
          href: {
            label: 'leggi',
            link: 'https://google1.it',
            title: 'vai al sito',
          },
          img: {
            alt: 'A picture showing some books',
            src: 'static/media/newsroom-cover.59549252.png',
          },
          title:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
        },
        {
          date: {
            date: new Date('2023-11-23T11:35:50.763Z'),
          },
          href: {
            label: 'leggi',
            link: 'https://google2.it',
            title: 'vai al sito',
          },
          img: {
            alt: 'A picture showing some books',
            src: 'static/media/newsroom-cover.59549252.png',
          },
          title:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
        },
        {
          date: {
            date: new Date('2023-11-23T11:35:50.763Z'),
          },
          href: {
            label: 'leggi',
            link: 'https://google3.it',
            title: 'vai al sito',
          },
          img: {
            alt: 'A picture showing some books',
            src: 'static/media/newsroom-cover.59549252.png',
          },
          title:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
        },
        {
          date: {
            date: new Date('2023-11-23T11:35:50.763Z'),
          },
          href: {
            label: 'leggi',
            link: 'https://google4.it',
            title: 'vai al sito',
          },
          img: {
            alt: 'A picture showing some books',
            src: 'static/media/newsroom-cover.59549252.png',
          },
          title:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
        },
        {
          date: {
            date: new Date('2023-11-23T11:35:50.763Z'),
          },
          href: {
            label: 'leggi',
            link: 'https://google5.it',
            title: 'vai al sito',
          },
          img: (
            <div>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
              eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing
              elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet,
              consectetur adipiscing elit, sed do eiusmod tempor.
            </div>
          ),
          title:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. ',
        },
      ]}
    />
  );
  expect(getAllByText('leggi')[0]?.closest('a')).toHaveAttribute(
    'href',
    'https://google1.it'
  );
  expect(getAllByText('leggi')[1]?.closest('a')).toHaveAttribute(
    'href',
    'https://google2.it'
  );
  expect(getAllByText('leggi')[2]?.closest('a')).toHaveAttribute(
    'href',
    'https://google3.it'
  );
  expect(getAllByText('leggi')[3]?.closest('a')).toHaveAttribute(
    'href',
    'https://google4.it'
  );
  expect(getAllByText('leggi')[4]?.closest('a')).toHaveAttribute(
    'href',
    'https://google5.it'
  );
});
