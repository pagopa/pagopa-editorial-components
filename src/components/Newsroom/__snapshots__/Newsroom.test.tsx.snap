// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly standard  1`] = `
<section
  className="MuiBox-root css-1epl6ta"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-3 css-ot7hnh-MuiGrid-root"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-9 css-neb4x5-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1reclri-MuiGrid-root"
          >
            <div
              className="MuiBox-root css-5a0uun"
            >
              <img
                alt="A picture showing some books"
                src="static/media/newsroom-cover.59549252.png"
                width="100%"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ct4b4t-MuiTypography-root"
            >
              23 novembre 2023
            </p>
            <h6
              className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. 
            </h6>
            <div
              className="MuiStack-root css-uqhr1i-MuiStack-root"
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1z0f9jx-MuiTypography-root-MuiLink-root"
                href="https://google.it"
                onBlur={[Function]}
                onFocus={[Function]}
                title="vai al sito"
              >
                leggi
              </a>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bnbs7j-MuiSvgIcon-root"
                data-testid="ArrowRightAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1reclri-MuiGrid-root"
          >
            <div
              className="MuiBox-root css-5a0uun"
            >
              <img
                alt="A picture showing some books"
                src="static/media/newsroom-cover.59549252.png"
                width="100%"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ct4b4t-MuiTypography-root"
            >
              23 novembre 2023
            </p>
            <h6
              className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. 
            </h6>
            <div
              className="MuiStack-root css-uqhr1i-MuiStack-root"
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1z0f9jx-MuiTypography-root-MuiLink-root"
                href="https://google.it"
                onBlur={[Function]}
                onFocus={[Function]}
                title="vai al sito"
              >
                leggi
              </a>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bnbs7j-MuiSvgIcon-root"
                data-testid="ArrowRightAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1reclri-MuiGrid-root"
          >
            <div
              className="MuiBox-root css-5a0uun"
            >
              <img
                alt="A picture showing some books"
                src="static/media/newsroom-cover.59549252.png"
                width="100%"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ct4b4t-MuiTypography-root"
            >
              23 novembre 2023
            </p>
            <h6
              className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. 
            </h6>
            <div
              className="MuiStack-root css-uqhr1i-MuiStack-root"
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1z0f9jx-MuiTypography-root-MuiLink-root"
                href="https://google.it"
                onBlur={[Function]}
                onFocus={[Function]}
                title="vai al sito"
              >
                leggi
              </a>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bnbs7j-MuiSvgIcon-root"
                data-testid="ArrowRightAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1reclri-MuiGrid-root"
          >
            <div
              className="MuiBox-root css-5a0uun"
            >
              <img
                alt="A picture showing some books"
                src="static/media/newsroom-cover.59549252.png"
                width="100%"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ct4b4t-MuiTypography-root"
            >
              23 novembre 2023
            </p>
            <h6
              className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. 
            </h6>
            <div
              className="MuiStack-root css-uqhr1i-MuiStack-root"
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1z0f9jx-MuiTypography-root-MuiLink-root"
                href="https://google.it"
                onBlur={[Function]}
                onFocus={[Function]}
                title="vai al sito"
              >
                leggi
              </a>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bnbs7j-MuiSvgIcon-root"
                data-testid="ArrowRightAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1reclri-MuiGrid-root"
          >
            <div
              className="MuiBox-root css-5a0uun"
            >
              <div>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor.
              </div>
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ct4b4t-MuiTypography-root"
            >
              23 novembre 2023
            </p>
            <h6
              className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor. 
            </h6>
            <div
              className="MuiStack-root css-uqhr1i-MuiStack-root"
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1z0f9jx-MuiTypography-root-MuiLink-root"
                href="https://google.it"
                onBlur={[Function]}
                onFocus={[Function]}
                title="vai al sito"
              >
                leggi
              </a>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-bnbs7j-MuiSvgIcon-root"
                data-testid="ArrowRightAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
